src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10374,21 +10384,28 @@ private static int GetNumberOfNonOmittedArraySizes(ArrayRankSpecifierSyntax rank
             return result;
         }
 
-        private bool IsPossibleArrayCreationExpression()
+        private bool IsTargetTypedNew(out bool isPossibleArrayCreation)
         {
             // previous token should be NewKeyword
 
+            ScanTypeFlags scanTypeFlags;
             var resetPoint = this.GetResetPoint();
             try
             {
-                ScanTypeFlags isType = this.ScanNonArrayType();
-                return isType != ScanTypeFlags.NotType && this.CurrentToken.Kind == SyntaxKind.OpenBracketToken;
+                scanTypeFlags = this.ScanNonArrayType();
+                isPossibleArrayCreation =
+                    scanTypeFlags != ScanTypeFlags.NotType &&
+                    this.CurrentToken.Kind == SyntaxKind.OpenBracketToken;
             }
             finally
             {
                 this.Reset(ref resetPoint);
                 this.Release(ref resetPoint);
             }
+
+            return !isPossibleArrayCreation &&
+                scanTypeFlags != ScanTypeFlags.NullableType &&
```

Comment:
Why the case for nullable types?
I expected a special case for tuples instead (since they start with open paren). How do we avoid returning `true` for `new (int, int) ...` here?

Reply:
>Why the case for nullable types?

because we still need to parse `new (Int32, Int32)?()`

>How do we avoid returning true for new (int, int) ... here?

`new (int, int) ...` is already forbidden, and now it's parsed as an argument list. we expect an expression instead of a type so it now produce two errors instead of a `ERR_NewWithTupleTypeSyntax`.